### PR TITLE
fix: accept uint8arrays

### PR DESCRIPTION
**Pull request by mikeal**
still a work in progress as i port dagdb to the latest multiformats stack.

### CODE REVIEW

**Comment by mikeal** [comment]
quick hack but it’s working now. ready for review.

File: iamap.js [hunk 1]
```diff
@@ -198,7 +198,8 @@ class IAMap {
     const hashBytes = hasherRegistry[options.hashAlg].hashBytes
 
     if (map !== undefined && !Buffer.isBuffer(map)) {
-      throw new TypeError('`map` must be a Buffer')
+      if (map instanceof Uint8Array) map = Buffer.from(map)
```
Comment (rvagg) [inline iamap.js:201]: style nit: always-braces and newlines with `if` and `else`

File: iamap.js [hunk 2]
```diff
@@ -594,6 +595,8 @@ function findElement (node, bitpos, key) {
   if (element.bucket) { // data element
     for (let bucketIndex = 0; bucketIndex < element.bucket.length; bucketIndex++) {
       const bucketEntry = element.bucket[bucketIndex]
+      const _key = bucketEntry.key
+      if (!Buffer.isBuffer(_key) && _key instanceof Uint8Array) bucketEntry.key = Buffer.from(_key)
```
Comment (rvagg) [inline iamap.js:599]: braces and newline

**Comment by alanshaw** [comment]
I just ran into this problem - please can it merge?

**Comment by rvagg** [comment]
done, 0.8.0, and I'll work on a 1.0.0 that removes `Buffer` entirely

**Comment by rvagg** [comment]
see https://github.com/rvagg/iamap/pull/11 for a complete switch to `Uint8Array`. use v1.0.0 to get that.

**Pull request closed without merging**